lib/puppet/application/ssl.rb

```diff
@@ -18,7 +18,7 @@ def help
 
 USAGE
 -----
-puppet ssl <action> [--certname <NAME>]
+puppet ssl <action> [--certname <NAME>] [--local-ca]
```

Comment:
I think this should be an underscore, since all the command names use underscores.

Reply:
yeah, we mostly cram multiple words into one (without dashes or underscores), eg `waitforcert`, `logdest`, `fromdate`, `sourceaddress`, `loadclasses`, `parseonly`, ...

We also use dashes:

```
$ git --no-pager grep ' option("--\w*-' lib/puppet/application/*.rb
lib/puppet/application/agent.rb:  option("--detailed-exitcodes") do |arg|
lib/puppet/application/agent.rb:  option("--job-id ID") do |arg|
lib/puppet/application/apply.rb:  option("--use-nodes")
lib/puppet/application/apply.rb:  option("--detailed-exitcodes")
lib/puppet/application/apply.rb:  option("--write-catalog-summary")
lib/puppet/application/device.rb:  option("--detailed-exitcodes") do |arg|
lib/puppet/application/face_base.rb:  option("--render-as FORMAT") do |format|
```

We do use underscores in two places:

```
$ git --no-pager grep ' option("--\w*_' lib/puppet/application/*.rb
lib/puppet/application/device.rb:  option("--to_yaml","-y")
lib/puppet/application/resource.rb:  option("--to_yaml","-y")
```

However all puppet settings can be specified on the command line e.g. `--dns_alt_names`, so we always use underscores there.

So I think using dashes makes sense for an action's option.... But I agree it's confusing having underscores and dashes... so what about shortening the action names, e.g. `puppet ssl download`, `puppet ssl submit`, `puppet ssl clean --local-ca`? The app would still accept `download_cert` and `submit_csr`, but the help would only mention the shorter version. Thoughts?

Reply:
Also how about `puppet ssl request` instead of `submit`?